esp/services/ldapenvironment/ldapenvironmentService.cpp

```diff
@@ -178,6 +178,47 @@ void CldapenvironmentEx::createLDAPBaseDN(const char * baseDN, SecPermissionType
     }
 }
 
+bool CldapenvironmentEx::createUser(StringBuffer &userName, const char * prefix, const char * envName, const char * baseDN, const char * pwd, StringBuffer &notes)
+{
+#define MAX_USERNAME_LDAP_LEN 20
+    StringBuffer tmpUserName;
+    tmpUserName.appendf("%s%s", prefix, envName);
+    userName.set(tmpUserName);
+
+    Owned<ISecUser> user = secmgr->createUser(tmpUserName.str());
+    user->credentials().setPassword(pwd);
+
+    int idx = 0;
+    while (true)
+    {
+        try
+        {
+            DBGLOG("Creating user '%s' in '%s', len=%d", tmpUserName.str(), baseDN, tmpUserName.length());
+            secmgr->addUser(*user.get(), baseDN);
+            userName.set(tmpUserName.str());//success!
+            return true;
+        }
+        catch(...)
+        {
+            if (idx == 99)
+            {
+                notes.appendf("\nNon Fatal Error creating '%s'", userName.str());
+                return false;
+            }
+            if (idx == 0 && tmpUserName.length() > MAX_USERNAME_LDAP_LEN)
+                tmpUserName.setLength(MAX_USERNAME_LDAP_LEN);//LDAP limits account names to 20 characters
```

Comment:
Is the advantage of doing this here, rather than from the very beginning, in case the size limit gets removed at the server at some point?  That is, longer names just might start working so respond to the error rather than shorten pro-actively?

Reply:
From what I read, this limit could be a domain policy (which is likely the default domain policy), and they imply that this restriction is only on older versions of AD.  So I figure, why not at least give it a try...